src/iota/modes/emacs.rs

```diff
@@ -0,0 +1,151 @@
+use keyboard::Key;
+use keymap::{KeyMap, KeyMapState};
+use buffer::Mark;
+use command::{BuilderEvent, Operation, Instruction, Command, Action};
+use textobject::{Anchor, Kind, TextObject, Offset};
+
+use super::Mode;
+
+
+/// Emacs mode uses Emacs-like keybindings.
+///
+pub struct EmacsMode {
+    keymap: KeyMap<Command>,
+    match_in_progress: bool,
+}
+
+impl EmacsMode {
+
+    /// Create a new instance of EmacsMode
+    pub fn new() -> EmacsMode {
+        EmacsMode {
+            keymap: EmacsMode::key_defaults(),
+            match_in_progress: false,
+        }
+    }
+
+    /// Creates a KeyMap with default EmacsMode key bindings
+    fn key_defaults() -> KeyMap<Command> {
+        let mut keymap = KeyMap::new();
+
+        // Editor Commands
+        keymap.bind_key(Key::Ctrl('q'), Command::exit_editor());
+        keymap.bind_key(Key::Ctrl('s'), Command::save_buffer());
```

Comment:
Is this is going to be more like emacs, then I think ctrl-q and ctrl-s can be removed.